src/js/main.js

```diff
@@ -0,0 +1,611 @@
+    const containerButtons = document.createElement("div");
+    containerButtons.append(inputClose, inputSave);
+    containerButtons.classList.add("modal-edit__container-btn");
+    //ADD TO FORM
+    formBody.append(labelTitle, labelBody);
+    //ADD FORM TO CONTAINER
+    parentContainer.append(backgorundImage, formBody, containerButtons);
+
+    inputSave.addEventListener("click", (e) => {
+        e.preventDefault();
+        modifyPost(e);
+    });
+
+    parentContainer.parentElement.classList.toggle("container--hide");
+}
+
+//!UPDATE INPUT FIELDS
+function updateValue(e) {
+    if ((e.target.id === "titleInput")) {
+        const title = document.getElementById("titleInput");
+        title.value = e.target.value;
+    } else if ((e.target.id === "bodyInput")) {
+        const body = document.getElementById("bodyInput");
+        body.value = e.target.value;
+    }
+}
+
+//!MODIFY POST BY FETCH REQUE
+//!GET DATA FOR MODAL SPECIFIC POST
+async function modifyPost(e) {
+    const popupModal = document.getElementById("popup");
+    const clickPostID = e.target.parentElement.previousSibling.dataset.id;
+    const urlPost = `http://localhost:3000/posts/${clickPostID}`;
+    const response = await fetch(urlPost);
+    const post = await response.json();
+    const result = await fetchToModifyPosts(urlPost, post);
+    //show popup info after modify post
+    popup(clickPostID, popupModal, true, result);
+}
+
+async function fetchToModifyPosts(url, post) {
+    const titlePost = document.getElementById("titleInput").value;
+    const bodyPost = document.getElementById("bodyInput").value;
+    try {
+        await fetch(url, {
+            method: 'PUT',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                userId: post.userId,
+                title: titlePost,
+                body: bodyPost,
+                id: post.id
+            })
+        });
+        updateCard(post.id, titlePost);
+        return false;
+    } catch (error) {
+        console.log(error);
+        return true;
+    }
+}
+
+//!FUNCTION TO REMOVE POST
+function removeCard(id) {
+    const element = document.querySelector((`article[data-id='${id}']`));
+    element.remove();
+}
+
+//!FUNCTION TO UPDATE POST CARD INFO
+function updateCard(id, title) {
+    const element = document.querySelector((`article[data-id='${id}']`));
+    const titleElement = element.querySelector(".card-title");
+    titleElement.textContent = title;
+}
+
+//!DELETE POST BY FETCH REQUEST
+async function deletePost(e) {
+    const clickPostID = e.target.parentElement.parentElement.parentElement.dataset.id;
+    const urlPost = `http://localhost:3000/posts/${clickPostID}`;
+    try {
+        await fetch(urlPost, {
+            method: 'DELETE'
+        });
+        // const result = await responseDELETE.text();
+        return false;
+    } catch (error) {
+        console.log(error);
```

Comment:
We have used console.error(error), but we don't know what should be a better practice